### PR TITLE
added uart_irq_dis/uart_irq_en to clear buffer

### LINKED ISSUE #1954: Characters get lost while copying a long (more then 7 symbols) string to the console
To reproduce the problem you can
* run x86/qemu or arm/qemu template
* Copy to clipboard "1234567890"
* Paste to Embox command line

You see "12345678" instead of  "1234567890"

### PR DESCRIPTION
**Pull request by sobhaniiest**
Closes #1954 
Thanks @anton-bondarev

### CODE REVIEW

File: src/drivers/tty/serial/ttys_processing.c [hunk 1]
```diff
@@ -41,23 +41,30 @@ static inline struct uart *tty2uart(struct tty *tty) {
 	return tu->uart;
 }
 
-static int uart_rx_buff_put(struct uart *dev, int c) {
+static int uart_rx_buff_put(unsigned int irq_nr, struct uart *dev) {
```
Comment (anton-bondarev) [inline src/drivers/tty/serial/ttys_processing.c:44]: Is 'irq_nr' used here?

File: src/drivers/tty/serial/ttys_processing.c [hunk 2]
```diff
@@ -41,23 +41,30 @@ static inline struct uart *tty2uart(struct tty *tty) {
...
-	{
-		rx = pool_alloc(&uart_rx_buff);
-		if (!rx) {
+	if (dev->tty) {
+		while (uart_hasrx(dev)) {
+			irq_lock();
+			{
+				rx = pool_alloc(&uart_rx_buff);
+				if (!rx) {
+					dev->uart_ops->uart_irq_dis(dev, &dev->params);
```
Comment (anton-bondarev) [inline src/drivers/tty/serial/ttys_processing.c:53]: You must check whether dev->uart_ops->uart_irq_dis exist or NULL!

File: src/drivers/tty/serial/ttys_processing.c (hunk 1 above)
Comment (sobhaniiest) [reply src/drivers/tty/serial/ttys_processing.c:44]: No! actually I kept it as it was in `uart_irq_handler`. I'll remove it.

File: src/drivers/tty/serial/ttys_processing.c (hunk 2 above)
Comment (sobhaniiest) [reply src/drivers/tty/serial/ttys_processing.c]: Okay! done.

File: src/drivers/tty/serial/ttys_processing.c [hunk 3]
```diff
@@ -41,23 +41,32 @@ static inline struct uart *tty2uart(struct tty *tty) {
...
-		if (!rx) {
+	if (dev->tty) {
+		while (uart_hasrx(dev)) {
+			irq_lock();
+			{
+				rx = pool_alloc(&uart_rx_buff);
+				if (!rx) {
+					if (dev->uart_ops->uart_irq_dis) {
+						dev->uart_ops->uart_irq_dis(dev, &dev->params);
+						lthread_launch(&uart_rx_irq_handler);
```
Comment (anton-bondarev) [inline src/drivers/tty/serial/ttys_processing.c:55]: ` lthread_launch(&uart_rx_irq_handler); ` have to called regardless of whether there is uart_irq_dis handler or not

Comment (sobhaniiest) [reply src/drivers/tty/serial/ttys_processing.c]: done

**Comment by anton-bondarev** [comment]
@sobhaniiest Please don't rebase during review, just commit ahead your branch.

**Comment by sobhaniiest** [comment]
Thanks!

**Pull request merged**